### PR TITLE
CI: Add python 3.7 support

### PR DESCRIPTION
**Pull request by kszucs**
resolves https://github.com/ibis-project/ibis/issues/1539

### CODE REVIEW

**Comment by kszucs** [comment]
Let's see whether the 3.6 dependencies will work...

**Comment by cpcloud** [comment]
@kszucs You will need #1543 for this to pass.

File: ci/requirements-dev-3.7.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - pytest-xdist
+  - python=3.7
+  - python-graphviz
+  - python-hdfs>=2.0.16
+  - pytz
+  - regex
+  - requests
+  - ruamel.yaml
+  - six
+  - sqlalchemy>=1.0.0,<1.1.15
```
Comment (cpcloud) [inline ci/requirements-dev-3.7.yml]: Packages that satisfy this constraint do not exist on `conda-forge`. After #1543 we'll be able to use `sqlalchemy` without constraints. There might still be additional work to do to get 3.7 working, we'll see.

Comment (kszucs) [reply ci/requirements-dev-3.7.yml]: Guessed so :)

**Comment by cpcloud** [comment]
@kszucs Can you rebase and push again?

**Comment by cpcloud** [comment]
@kszucs This might be a little easier though there is this issue to be aware of: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5637

**Comment by kszucs** [comment]
@cpcloud We have four python 3.7 incompatible [dependencies](https://github.com/ibis-project/ibis/pull/1544/files#diff-1ff682cf0a7299a88ec03b297b5f4c43), I've created a PR for each of them.

**Comment by cpcloud** [comment]
@kszucs Great! Thanks for doing that. We should still be able to run tests without those deps (though we probably can't build conda packages).

**Comment by kszucs** [comment]
`lz4` and `ruamel.yaml` are just waiting for jakirkham to merge, so those will probably resolved today.

We can handle `pyarrow`, however theres is a dependency loop [again](https://circleci.com/gh/conda-forge/pyarrow-feedstock/202), so that will take a couple of hours to properly build them.

**Comment by kszucs** [comment]
@cpcloud theoretically We're good to go on python 3.7 now, there are strange Postgre issues

**Comment by cpcloud** [comment]
https://github.com/cloudera/impyla/issues/312 is currently blocking this.

**Comment by cpcloud** [comment]
The tests are probably able to run because we are careful about making sure we call `pytest.importorskip` correctly.

**Comment by cpcloud** [comment]
Hm, actually I don't know how the tests are even able to start

**Comment by kszucs** [comment]
resolved by #1696

**Pull request closed without merging**